### PR TITLE
improved filter design

### PR DESCRIPTION
**Pull request by VishakhaSainani**
fixes : #141 

Screenshots: 
React Quiz Component Demo - Google Chrome 23-10-2023 23_57_26
React Quiz Component Demo - Google Chrome 23-10-2023 23_57_42

I used div , li and ul were giving lint errors. Check this out.

### CODE REVIEW

**Comment by VishakhaSainani** [comment]
@wingkwong Please review it.

**Review by wingkwong — CHANGES_REQUESTED** [review]
1. the filter is not working at all...
2. the new filter is a bit large. You may refer to [ant design select](https://ant.design/components/select) - the first one in basic usage. It's close to my expectation.
3. wrap the style by `.react-quiz-container`. i.e. `.react-quiz-container .quiz-result-filter`
4. need a better name for `custom-select`

**Comment by VishakhaSainani** [comment]
@wingkwong Requested changes updated please check.

**Comment by wingkwong** [comment]
> Can you please elaborate this one, of what item exactly the height needs to be changed. The select button or the cursor arrow or the dropdown menu.

The overall height. Now it's 42px. Change to 32px. The text and arrow down icon can be a bit smaller.

image

**Comment by VishakhaSainani** [comment]
@wingkwong Check it out. Thankyou.

File: src/lib/styles.css [hunk 1]
```diff
@@ -171,4 +151,102 @@
...
   display: flex;
   justify-content: space-between;
-}
\ No newline at end of file
+}
+
+.react-quiz-container .quiz-result-filter {
+  position: relative;
+  width: 120px;
+  margin-top: -18px;
```
Comment (wingkwong) [inline src/lib/styles.css:159]: don't use negative value. It's a bad practice.

File: src/lib/styles.css [hunk 2]
```diff
@@ -171,4 +151,102 @@
...
+  .filter-dropdown-select {
+    padding: 8px;
+    text-transform: capitalize;
+    border: 2px solid #c2c5c7;
+    border-radius: 7px;
+    background-color: #fcfcfc;
+    color: #333;
+    font-size: 16px;
+    cursor: pointer;
+    margin-bottom: -5px;
```
Comment (wingkwong) [inline src/lib/styles.css:171]: same as above

File: src/lib/styles.css [hunk 3]
```diff
@@ -171,4 +151,102 @@
...
+    bottom: 10px;
+    position: relative; 
+    height: 12.45px;
```
Comment (wingkwong) [inline src/lib/styles.css:174]: shouldn't set decimal.

File: src/lib/styles.css [hunk 4]
```diff
@@ -171,4 +151,102 @@
...
+  }
+
+  .dropdown-options .dropdown-options-item:focus{
+    background-color: #d7f1f9;
+    font-weight: bold;
+  }
+
+  .filter-dropdown-select .selected-option.selected-open {
+    color: #c6c2c2; 
+    font-weight:lighter;
```
Comment (wingkwong) [inline src/lib/styles.css:230]: make sure there is a space after `:` (apply to all)

File: src/lib/styles.css [hunk 5]
```diff
@@ -171,4 +151,102 @@
...
+    font-size: 16px;
+    cursor: pointer;
+  }
+
+  .dropdown-options .dropdown-options-item:hover {
+    background-color: rgba(226, 226, 226, 0.852);
+    border-radius: 5px;
+  }
+
+  .dropdown-options .dropdown-options-item:focus{
```
Comment (wingkwong) [inline src/lib/styles.css:223]: make sure there is a space after `{` (apply to all)

**Review by wingkwong — COMMENTED** [review]
- open a select option list, unfocus should close the list. 
- the font size of selected option and that of options in the list are not aligned
- the gap between the select and the list should be reduced a bit
image

**Comment by VishakhaSainani** [comment]
@wingkwong I hope it's good enough now, check it out. Thankyou

**Review by wingkwong — APPROVED** [review]

**Pull request merged**